fastiuk/sonic-swss-common
# Login messages config

This PR depends on https://github.com/fastiuk/sonic-buildimage/pull/4

#### Why I did it
* Add an ability to configure login messages: /etc/motd, /etc/issue*
* Be able to use table defines from swsscommon

#### How to verify it
Build SONiC
or
When SONiC was already built:
```
rm -f target/python-wheels/bullseye/sonic_yang_models-1.0-py3-none-any.whl
BLDENV=bullseye make -f Makefile.work target/python-wheels/bullseye/sonic_yang_models-1.0-py3-none-any.whl
rm -f target/python-wheels/bullseye/sonic_host_services-1.0-py3-none-any.whl
BLDENV=bullseye make -f Makefile.work target/python-wheels/bullseye/sonic_host_services-1.0-py3-none-any.whl
```
Login into the switch.
Set messages:
```
redis-cli -n 4 HSET "LOGIN_MESSAGE|pre_login" "message" "Test pre login"
redis-cli -n 4 HSET "LOGIN_MESSAGE|post_login" "message" "Test post login"
```
Logout/login and see the results:
<img width="829" alt="Screenshot 2022-07-02 at 22 58 11" src="https://user-images.githubusercontent.com/35844154/177014657-2cd65bf6-19b1-4b6d-b989-1817582870dd.png">
Clear messages:
```
redis-cli -n 4 HSET "LOGIN_MESSAGE|pre_login" "message" ""
redis-cli -n 4 HSET "LOGIN_MESSAGE|post_login" "message" ""
```

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog
* Add an ability to configure pre-login and post-login messages: /etc/motd, /etc/issue*

#### Link to config_db schema for YANG module changes
[sonic-login_message.yang](https://github.com/fastiuk/sonic-buildimage/tree/login-message-config/src/sonic-yang-models/yang-models/sonic-login_message.yang)

#### Superproject PR:
| PR title | State |
| ----------------- | ----------------- |
| [Login messages config](https://github.com/fastiuk/sonic-buildimage/pull/4) | ![GitHub issue/pull request detail](https://img.shields.io/github/pulls/detail/state/fastiuk/sonic-buildimage/4) |

#### A picture of a cute animal
![1-46](https://user-images.githubusercontent.com/35844154/169010699-e61cb22f-c9ae-4d12-8ef1-64e3607eef4c.jpeg)
